tracking/src/main/java/org/hps/recon/tracking/NearestNeighborRMSClusterer.java

```diff
@@ -228,14 +275,185 @@ public List<List<LCRelation>> findClusters(List<LCRelation> fittedHits) {
             // Finished with this cluster, check cluster threshold and add it to the list of
             // clusters
             if (cluster.size() > 0 && cluster_signal / Math.sqrt(cluster_noise_squared) > _cluster_threshold) {
-                cluster_list.add(cluster);
+		if(!(_doVSplit)||cluster.size()<=2){cluster_list.add(cluster);}
+		else{
+	    	    for(int j=0;j<cluster.size();j++){
```

Comment:
Clean up this for loop, it isn't actually doing anything beyond using memory

Reply:
done